app/controllers/merchant_items_controller.rb

```diff
@@ -6,13 +6,19 @@ def index
   end
 
   def update
-    @merchant = Merchant.find(params[:merchant_id])
+    merchant = Merchant.find(params[:merchant_id])
+    item = merchant.items.find(params[:item_id])
 
     if params[:change_item_status]
-      @item = @merchant.items.find(params[:item_id_for_change])
-      @item.change_status
+      item.change_status
```

Comment:
we could probably dry up the code here.  somehow merge change_item_status and other updates.  But seeing how I am still pretty lost with API, we should probably leave it alone.  lol.